### PR TITLE
csi with openshift support

### PR DESCRIPTION
**Pull request by karmab**
**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#################################################################################################################
+# The deployment for the rook operator that enables the ceph-csi driver for beta testing.
+# For example, to create the rook-ceph cluster:
+#   kubectl create -f common.yaml
+#   kubectl create -f operator-with-csi.yaml
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml]: this line needs to be updated with the new filename

File: cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        - name: ROOK_CSI_PROVISIONER_IMAGE
+          value: "quay.io/k8scsi/csi-provisioner:v1.0.1"
+        - name: ROOK_CSI_SNAPSHOTTER_IMAGE
+          value: "quay.io/k8scsi/csi-snapshotter:v1.0.1"
+        - name: ROOK_CSI_ATTACHER_IMAGE
+          value: "quay.io/k8scsi/csi-attacher:v1.0.1"
+        - name: ROOK_HOSTPATH_REQUIRES_PRIVILEGED
+          value: "true"
+        - name: FLEXVOLUME_DIR_PATH
+          value: "/etc/kubernetes/kubelet-plugins/volume/exec"
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml]: should we set this FLEXVOLUME_DIR_PATH also in `operator-openshift.yaml`?

Comment (karmab) [reply cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml]: i believe we should indeed

Comment (travisn) [reply cluster/examples/kubernetes/ceph/operator-openshift-with-csi.yaml]: I knew I saw that somewhere :)

**Comment by travisn** [comment]
@karmab Would you mind adding a description for this new example to the [help doc](https://github.com/rook/rook/blob/master/Documentation/ceph-examples.md#operator)? thanks!

**Comment by karmab** [comment]
> help doc

done!

File: Documentation/ceph-examples.md [hunk 3]
```diff
@@ -34,6 +34,9 @@ After the common resources are created, the next step is to create the Operator
    - `oc create -f operator-openshift.yaml`
 - `operator-with-csi.yaml`: Includes configuration for testing ceph-csi while the integration is still in progress. See the [CSI Drivers](ceph-csi-drivers.md) topic for more details.
    - `kubectl create -f operator-with-csi.yaml`
+   - `kubectl create -f operator-with-csi.yaml`
```
Comment (travisn) [inline Documentation/ceph-examples.md]: Looks like a duplicate line.

**Comment by travisn** [comment]
Could you also add a PR description above? Thanks!

**Comment by BlaineEXE** [comment]
It seems like we may be getting to the point where we should add an `openshift` directory next to `kubernetes` and `coreos`. The `kubernetes/ceph` dir is starting to feel overfull to me. Another option would be to put them in `kubernetes/ceph/openshift`.

**Comment by travisn** [comment]
When refactoring the yamls that question also crossed my mind. If different versions for all (or at least majority) of the yamls were needed for a scenario, then a separate folder would certainly be a good thing. But if only a handful are different then it is a pain either to work in multiple directories or to keep symlinks updated in the subdir. 

I don't have a strong opinion against the subdirectory for OpenShift, but my impression was that there weren't enough different yamls to justify it.

**Pull request merged**